numpy/core/tests/test_dtype.py

```diff
@@ -445,6 +446,54 @@ def test_alignment(self):
         t2 = np.dtype('2i4', align=True)
         assert_equal(t1.alignment, t2.alignment)
 
+    @pytest.mark.skipif(not HAS_REFCOUNT, reason="Python lacks refcounts")
+    def test_structured_with_object(self):
+        # Test that object creation and access works for complex dtypes
+        # with subarrays.
+        dt1 = np.dtype([('a', 'i4'),
+                        ('b', [('ba', 'O'), ('bb', 'i1')], (2, 3))])
+        dt2 = np.dtype([('a', 'i4'),
+                        ('b', [('ba', 'O'), ('bb', 'O')], (2, 3))])
+        dt3 = np.dtype([('a', 'i4'), ('b', 'O', (2, 3))])
+
+        # create patterns to fill array manually:
+        singleton = object()
+        p1 = (0, [[(singleton, 0)] * 3] * 2)
+        p2 = (0, [[(singleton, singleton)] * 3] * 2)
+        p3 = (0, [[singleton] * 3] * 2)
+
+        one = 1
+        zero = 0
+        for dt, count, pat in zip([dt1, dt2, dt3], [6, 12, 6], [p1, p2, p3]):
+            # Test assumes that 0, 1, and None are singletons.
+            for func, obj in zip([np.empty, np.ones, np.zeros], [None, one, zero]):
+                if obj is None:
+                    # None test is flaky:
```

Comment:
Might be good to elaborate a bit on this comment: is it numpy or python that is flaky with `None`?

Reply:
Ooops, leftover for some earlier version, I think the test runs fine now.

Reply:
Maybe I should call None also not `None` here, I got a feeling it helped for the others. I was wondering if it might have to do with the actual internal code python compiles too...

Reply:
Doesn't help. It all seems to have to do with trickier things. Replacing the assert with a print for example runs it fine for me.

Frankly, I am not quite sure if the other tests (one and zero) are reliable or not. They seem to be, but..

Reply:
Yikes...